src/util.rs

```diff
@@ -32,3 +32,38 @@ impl Maximums {
         }
     }
 }
+
+#[macro_export]
+macro_rules! pg_enum {
+    (
+        $vis:vis enum $name:ident {
+            $($item:ident = $int:expr,)*
+        }
+    ) => {
+        #[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize, FromSqlRow, AsExpression)]
+        #[diesel(sql_type = diesel::sql_types::Integer)]
+        #[serde(rename_all = "snake_case")]
+        #[repr(i32)]
+        $vis enum $name {
+            $($item = $int,)*
+        }
+
+        impl diesel::deserialize::FromSql<diesel::sql_types::Integer, diesel::pg::Pg> for $name {
+            fn from_sql(bytes: diesel::pg::PgValue<'_>) -> diesel::deserialize::Result<Self> {
+                match <i32 as diesel::deserialize::FromSql<diesel::sql_types::Integer, diesel::pg::Pg>>::from_sql(bytes)? {
+                    $($int => Ok(Self::$item),)*
+                    n => Err(format!("unknown value: {}", n).into()),
```

Comment:
can we somehow integrate `$name` in the error message to make this easier to debug?

Reply:
Done!